tests/snuba/api/endpoints/test_organization_events_v2.py

```diff
@@ -5171,3 +5167,321 @@ def test_performance_homepage_query(self):
+        response = self.do_request(query)
+        assert response.status_code == 200, response.content
+        data = response.data["data"]
+        assert response.data["meta"]["isMetricsData"]
+        assert len(data) == 1
+        assert data[0]["team_key_transaction"] == 0
+        assert data[0]["transaction"] == "bar_transaction"
+
+        # not key transactions
+        query["query"] = "team_key_transaction:false"
+        response = self.do_request(query)
+        assert response.status_code == 200, response.content
+        data = response.data["data"]
+        assert response.data["meta"]["isMetricsData"]
+        assert len(data) == 1
+        assert data[0]["team_key_transaction"] == 0
+        assert data[0]["transaction"] == "bar_transaction"
+
+    def test_too_many_team_key_transactions(self):
+        MAX_QUERYABLE_TEAM_KEY_TRANSACTIONS = 1
+        with mock.patch(
+            "sentry.search.events.fields.MAX_QUERYABLE_TEAM_KEY_TRANSACTIONS",
+            MAX_QUERYABLE_TEAM_KEY_TRANSACTIONS,
+        ):
+            team = self.create_team(organization=self.organization, name="Team A")
+            self.create_team_membership(team, user=self.user)
+            self.project.add_team(team)
+            project_team = ProjectTeam.objects.get(project=self.project, team=team)
+            transactions = ["foo_transaction", "bar_transaction", "baz_transaction"]
+
+            for i in range(MAX_QUERYABLE_TEAM_KEY_TRANSACTIONS + 1):
+                self.store_metric(
+                    100, tags={"transaction": transactions[i]}, timestamp=self.min_ago
+                )
+
+            TeamKeyTransaction.objects.bulk_create(
+                [
+                    TeamKeyTransaction(
+                        organization=self.organization,
+                        project_team=project_team,
+                        transaction=transactions[i],
+                    )
+                    for i in range(MAX_QUERYABLE_TEAM_KEY_TRANSACTIONS + 1)
+                ]
+            )
+
+            query = {
+                "team": "myteams",
+                "project": [self.project.id],
+                "orderby": "p95()",
+                "field": [
+                    "team_key_transaction",
+                    "transaction",
+                    "transaction.status",
+                    "project",
+                    "epm()",
+                    "failure_rate()",
+                    "p95()",
+                ],
+                "metricsEnhanced": "1",
+                "per_page": 50,
+            }
+
+            response = self.do_request(query)
+            assert response.status_code == 200, response.content
+            data = response.data["data"]
+            assert len(data) == 2
+            assert response.data["meta"]["isMetricsData"]
+            assert (
+                sum(row["team_key_transaction"] for row in data)
+                == MAX_QUERYABLE_TEAM_KEY_TRANSACTIONS
+            )
```

Comment:
Code in asserts 😎 , I'm guessing it's tiring typing them out 
```py
        assert data[0]["team_key_transaction"] == 1
        ...
        assert data[n]["team_key_transaction"] == 1
```

Reply:
Hmm this was copy pasta but good point its only 2, probably should just type them all out